core/shared/src/main/scala/scalaz/zio/ZSchedule.scala

```diff
@@ -564,9 +575,9 @@ trait Schedule_Functions extends Serializable {
    * A new schedule derived from the specified schedule which adds the delay
    * specified as output to the existing duration.
    */
-  final def delayed[R: ConformsR, A](s: ZSchedule[R, A, Duration]): ZSchedule[R, A, Duration] = {
+  final def delayed[R: ConformsR, A](s: ZSchedule[R, A, Delay]): ZSchedule[R, A, Delay] = {
     val delayed = s.modifyDelay((b, d) => IO.succeed(b + d))
-    delayed.reconsider((_, step) => step.copy(finish = () => step.delay)) // TODO: Dotty doesn't infer this properly
+    delayed.reconsider((_, step) => step.copy(finish = () => step.delay))
```

Comment:
Please add the comment about Dotty, it's how we are tracking Dotty type inference regressions.